pandas/core/reshape/merge.py

```diff
@@ -533,33 +540,37 @@ def merge_asof(
 # TODO: only copy DataFrames when modification necessary
 class _MergeOperation:
     """
-    Perform a database (SQL) merge operation between two DataFrame objects
-    using either columns as keys or their row indexes
+    Perform a database (SQL) merge operation between two DataFrame or Series
+    objects using either columns as keys or their row indexes
     """
 
     _merge_type = "merge"
 
+    indicator_name: Optional[str]
+    left: "DataFrame"
```

Comment:
Is there a reason for adding these definitions at the class level? Would prefer not to do that

Reply:
there was a request to use py36 style annotations

Reply:
Can these be done in the __init__? These are instance variables shouldn't need to expose as class variables now for annotations

Reply:
i guess so, sure.  is this the wrong use case for putting these a the class level?  or is it a preference/policy thing?

Reply:
> Can these be done in the **init**? These are instance variables shouldn't need to expose as class variables now for annotations

since these are type declarations and not variable initialisations, I _think_ that these are a noop at runtime and therefore not exposed as class variables.

```python
>>> class foo():
...     bar: str
...
>>> foo().bar
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'foo' object has no attribute 'bar'
```

Reply:
Is there consensus on the desired usage?

Reply:
If it can be done in the init I still think better. Should keep things localized and not push to higher namespaces unless really necessary

Reply:
agreed.

Reply:
updated